### PR TITLE
Add ld_default package

### PR DESCRIPTION
**Pull request by uberhacker**
This package enables you to select the default ld executable.  The binutils package installs 2 different versions of ld (ld.bfd and ld.gold).  Sometimes the linker has trouble compiling certain source files and using the other version can help to resolve this problem.  Also, you may want to look in `#{CREW_LIB_PREFIX}` for libraries before `/usr/#{ARCH_LIB}` or `/#{ARCH_LIB}`.  Normally, libraries located in `/#{ARCH_LIB}` will be selected over `#{CREW_LIB_PREFIX}`.  This package lets you override the normal behavior.

### CODE REVIEW

**Comment by JL2210** [comment]
Could you also make `lld` the default linker with this?

**Comment by JL2210** [comment]
Also, this would overwrite the default linker, as the default linker is a symlink.

**Comment by JL2210** [comment]
`ld.lld` is the linker name itself. `lld` itself is just a program that states to use `ld.lld` on UNIX.

**Comment by JL2210** [comment]
Also, you used `ldd` instead of `lld`. I make that mistake a lot.

**Comment by JL2210** [comment]
You also DO NOT want to use libraries from `/usr/lib64`, `/usr/lib`, `/lib64`, or `/lib` on account of the fact that those directories contain libraries that have the `.relr.dyn` section. The GNU `ld` is not able to link against those. Anyway, those directories are by default in the dynamic linker search path, so no need. A useful option would be `-rpath #{CREW_LIB_PREFIX}`

**Comment by JL2210** [comment]
The package still overwrites the linker you chose to make default. I had to reinstall binutils to have `ld` work at all.

**Comment by uberhacker** [comment]
That is fine.  If you reinstall ld_default, you can also get the linker to work again.

**Comment by JL2210** [comment]
No, you can't. That's what I'm saying. For example, if you want `ld.gold` as default, it overwrites `ld.gold` with the script meant to be run by `ld`.

**Comment by uberhacker** [comment]
Not sure what you are talking about.  The only file that is overwrote is `ld`, but that is just a copy of `ld.bfd` when you install binutils.  You're still invoking the default linker you choose.  Try it again.

**Comment by JL2210** [comment]
`ld.bfd` after installing ld_default:
```
#!/bin/bash
/usr/local/bin/ld.bfd -L /usr/local/lib64 -L /usr/lib64 -L /lib64 "$@"
```

**Comment by JL2210** [comment]
I found the problem. You can't run it from `/usr/local/bin`. Look at my comments above and this is good.

**Comment by JL2210** [comment]
Other than the fact that a file called `ld` is created in the current directory.

**Comment by JL2210** [comment]
Why do we need this package anyway? Someone who wanted to could just as easily hardlink their favorite linker to `ld`.

File: packages/ld_default.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      end
+    end
+    system "echo '#!/bin/bash' > ld"
+    system "echo '#{CREW_PREFIX}/bin/#{$current} -L #{CREW_LIB_PREFIX} -L /usr/#{ARCH_LIB} -L /#{ARCH_LIB} \"$@\"' >> ld"
+    system "install -Dm755 ld #{CREW_PREFIX}/bin/ld"
+  end
+
+  def self.install
+    system "echo '#!/bin/bash' > ld"
+    system "echo '#{CREW_PREFIX}/bin/#{$current} -L #{CREW_LIB_PREFIX} -L /usr/#{ARCH_LIB} -L /#{ARCH_LIB} \"$@\"' >> ld"
```
[the remark this answers is not in the document]
Comment (uberhacker) [reply packages/ld_default.rb]: It's not about redundancy, it's about the ORDER the linker determines which libraries to link against.  Let's say I want to link using `/usr/local/lib/library.so` instead of the system `/lib/library.so`.  The way it works now is the system library will be chosen even though the desired library is `/usr/local/lib/library.so`.  If you don't believe me, try and compile vscode from source and see what happens.  I haven't noticed it much but sometimes `ld.gold` works when `ld.bfd` fails.  This seems to happen more often on i686 than any other architecture from my experience.

[the remark this answers is not in the document]
Comment (JL2210) [reply packages/ld_default.rb]: What? `ld.bfd` is much more stable than `ld.gold`. You really could just disable using any system libraries if you just use the `-rpath` option.

[the remark this answers is not in the document]
Comment (JL2210) [reply packages/ld_default.rb]: If you want to, just change the `--with-default-library-search-path=<blah:blah:blah>` and the `--enable-gold` options in `binutils`' configure options.

**Comment by uberhacker** [comment]
Not everybody is as linker savvy as you and knows how to symlink manually.  This package is mainly for convenience.  Not sure where you are getting your information about linkers.  The gold version is newer but not necessarily any less stable than the bfd version.

**Comment by JL2210** [comment]
If a linker cannot even build the Linux kernel, it's unstable.

**Comment by JL2210** [comment]
Can I fix this up for you? I think I could make it work much better.

**Comment by uberhacker** [comment]
How so?

**Comment by JL2210** [comment]
Let me work on this for a few minutes and show you.

**Comment by uberhacker** [comment]
In case you are wondering, you CANNOT symlink to ld.lld.  It won't work.

**Comment by JL2210** [comment]
Yeah, I know. Hardlinks only.

**Comment by uberhacker** [comment]
Also, symlinks/hardlinks or whatever won't let you override the options.  That is the most important reason this package was created.

**Comment by JL2210** [comment]
Yeah, give me a minute. I know.

**Comment by JL2210** [comment]
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      system "echo \"Current ld is hardlinked to $(find . -inum `ls -i ld | cut -d' ' -f1` | fgrep 'ld.' | cut -d/ -f2)\""
    end
    puts
    puts "Enter the ld default:"
    puts "b = ld.bfd"
    puts "g = ld.gold"
    puts "l = ld.lld"
    puts "0 = Cancel"

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $current = "ld.bfd"
        break
      when 'g'
        $current = "ld.gold"
        break
      when 'l'
        $current = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "Please select from one of the options or enter 0 to cancel."
      end
    end
    system "echo '#!/bin/bash' > ld"
    system "echo '#{$current} -rpath #{CREW_LIB_PREFIX} \"$@\"' >> ld"
  end

  def self.install
    system "install -Dm755 ld #{CREW_DEST_PREFIX}/bin/ld"
  end

  def self.postinstall
    puts
    puts "If you ever wish to change the default linker, simply run `crew remove ld_default && crew install ld_default`".lightblue
    puts
  end
end
```

**Comment by JL2210** [comment]
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      system "echo \"Current ld is hardlinked to $(find . -inum `ls -i ld | cut -d' ' -f1` | fgrep 'ld.' | cut -d/ -f2)\""
      $current = `find . -inum $(ls -i ld | cut -d' ' -f1) | fgrep 'ld.' | cut -d/ -f2`
    end
    puts
    puts "Enter the ld default:"
    puts "b = ld.bfd"
    puts "g = ld.gold"
    puts "l = ld.lld"
    puts "0 = Cancel"

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $current = "ld.bfd"
        break
      when 'g'
        $current = "ld.gold"
        break
      when 'l'
        $current = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "Please select from one of the options or enter 0 to cancel."
      end
    end
    system "echo '#!/bin/bash' > ld"
    system "echo '#{$current} -rpath #{CREW_LIB_PREFIX} \"$@\"' >> ld"
  end

  def self.install
    system "install -Dm755 ld #{CREW_DEST_PREFIX}/bin/ld"
  end

  def self.postinstall
    puts
    puts "If you ever wish to change the default linker, simply run `crew remove ld_default && crew install ld_default`".lightblue
    puts
  end
end
```

**Comment by JL2210** [comment]
@uberhacker, anything wrong with this?
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      $current = `find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' | cut -d/ -f2; `
      if "#{$current}" == "" then
        $current = "nothing"
      end
      system "echo \"Current default ld is hardlinked to #{$current}.\""
    end
    puts
    puts "Enter the new ld default:"
    puts "b = ld.bfd"
    puts "g = ld.gold"
    puts "l = ld.lld"
    puts "0 = Cancel"

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $newdefault = "ld.bfd"
        break
      when 'g'
        $newdefault = "ld.gold"
        break
      when 'l'
        $newdefault = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "Please select from one of the options or enter 0 to cancel."
      end
    end
    system "echo '#!/bin/bash' > ld"
    system "echo '#{$newdefault} -rpath #{CREW_LIB_PREFIX} \"$@\"' >> ld"
  end

  def self.install
    system "install -Dm755 ld #{CREW_DEST_PREFIX}/bin/ld"
  end

  def self.postinstall
    puts
    puts "If you ever wish to change the default linker again, simply run `crew remove ld_default && crew install ld_default`".lightblue
    puts
  end
end
```

**Comment by uberhacker** [comment]
As soon as the ld executable is overwrote by the bash shell script, your current logic will fail.  That algorithm is mainly there to find the current ld default after the fact, actually.  The first time this is run, it will return jibberish for the current default.

**Comment by JL2210** [comment]
Actually, the second time it is run, it will. You should really try it first. If the file does not exist or is not a hardlink, the script says `The current ld is hardlinked to nothing.`

**Comment by uberhacker** [comment]
That is the wrong logic.  I tell you what, I'll work on combining your logic so this is a bit more elegant.

**Comment by JL2210** [comment]
What do you mean, this is the wrong logic? How?

**Comment by JL2210** [comment]
It works well. If the commands that set `$current` fail and `"#{$current}" == ""`; then `$current` is set to `nothing`.

**Comment by uberhacker** [comment]
I don't want it ever to return 'nothing'.  I could give a rats @$$ about the hardlinking also.

**Comment by JL2210** [comment]
Did you read it? `binutils` hardlinks `ld` to either `ld.bfd` or `ld.gold` when it's installed.

**Comment by JL2210** [comment]
It's accurate. If `ld` is not a hardlink, then it merely states that it's not hardlinked.

**Comment by JL2210** [comment]
Improved version, @uberhacker. This version is more resilient to `ld` not being a hardlink:
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      $current = `find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' | cut -d/ -f2; `
      if "#{$current}" == "" then
        $current = "nothing"
      end
      if "#{$current}" == "nothing" then
        $current = `cat #{CREW_PREFIX}/bin/ld | tail -1 | cut -d' ' -f1`
      end
      if "#{$current}" == "nothing" then
        $current = `basename $(readlink #{CREW_PREFIX}/bin/ld)`
      end
      system "echo \"Current default ld is #{$current}\""
    end
    puts "Enter the new ld default:"
    puts "b = ld.bfd"
    puts "g = ld.gold"
    puts "l = ld.lld"
    puts "0 = Cancel"

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $newdefault = "ld.bfd"
        break
      when 'g'
        $newdefault = "ld.gold"
        break
      when 'l'
        $newdefault = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "Please select from one of the options or enter 0 to cancel."
      end
    end
    system "echo '#!/bin/bash' > ld"
    system "echo '#{$newdefault} -rpath #{CREW_LIB_PREFIX} \"$@\"' >> ld"
  end

  def self.install
    system "install -Dm755 ld #{CREW_DEST_PREFIX}/bin/ld"
  end

  def self.postinstall
    puts
    puts "If you ever wish to change the default linker again, simply run `crew remove ld_default && crew install ld_default`".lightblue
    puts
  end
end
```

**Comment by uberhacker** [comment]
Not sure where you are getting your information.  See below: (this is after a fresh binutils install)
```
chronos@localhost /usr/local/bin $ ls -l ld*
-rwxr-xr-x 4 chronos chronos  3039160 Jul 28 14:03 ld
lrwxrwxrwx 1 chronos chronos        3 Sep 30 07:47 ld64.lld -> lld
-rwxr-xr-x 4 chronos chronos  3039160 Jul 28 14:03 ld.bfd
-rwxr-xr-x 1 chronos chronos     5874 Jun 24 20:48 ldd
-rwxr-xr-x 1 chronos chronos    19100 Jun 24 20:48 lddlibc4
-rwxr-xr-x 2 chronos chronos 75628344 Jul 28 14:03 ld.gold
lrwxrwxrwx 1 chronos chronos        3 Sep 30 07:47 ld.lld -> lld
chronos@localhost /usr/local/bin $ diff ld ld.bfd
```
If ld was a hardlink, wouldn't u see an 'l' in the first column?  Also there is no difference between ld and ld.bfd which tells me it's a straight-up copy of the ld.bfd executable.

**Comment by JL2210** [comment]
No, a symlink shows `l` in the left column. A hardlink has the same inode as the file itt is linked to

**Comment by JL2210** [comment]
There should be no difference between hardlinked files other than the file name. They have the same inode numbers, so they are not copies, per se. https://medium.com/@wendymayorgasegura/what-is-the-difference-between-a-hard-link-and-a-symbolic-link-8c0493041b62

**Comment by JL2210** [comment]
Anyway, this should be the most fixed that it could possibly be at this point:
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'https://raw.githubusercontent.com/skycocker/chromebrew/93aa4dc4d9198e28d84dcf9d59b3ab6b6a94d92c/README.md'
  source_sha256 '3ec9c6f7ba6a138d07dfb4a372d562cd3ad558b58b6bd447c2dbb52e16b697a5'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      $current = `find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' | cut -d/ -f2`
      if "#{$current}" == "" then
        $current = "nothing"
      end
      if "#{$current}" == "nothing" then
        $current = `basename $(readlink #{CREW_PREFIX}/bin/ld)`
      end
      if "#{$current}" == "" then
        $current = "nothing"
      end
      if "#{$current}" == "nothing" then
        $current = `cat #{CREW_PREFIX}/bin/ld | tail -1 | cut -d' ' -f1`
      end
      system "echo \"Current default ld is #{$current}\""
    end
    puts "Enter the new ld default:"
    puts "b = ld.bfd"
    puts "g = ld.gold"
    puts "l = ld.lld"
    puts "0 = Cancel"

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $newdefault = "ld.bfd"
        break
      when 'g'
        $newdefault = "ld.gold"
        break
      when 'l'
        $newdefault = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "Please select from one of the options or enter 0 to cancel."
      end
    end
    system "echo '#!/bin/bash' > ld"
    system "echo '#{$newdefault} -rpath #{CREW_LIB_PREFIX} \"$@\"' >> ld"
  end

  def self.install
    system "install -Dm755 ld #{CREW_DEST_PREFIX}/bin/ld"
  end

  def self.postinstall
    puts
    puts "If you ever wish to change the default linker again, simply run `crew remove ld_default && crew install ld_default`".lightblue
    puts
  end
end
```

**Comment by JL2210** [comment]
In your case, `ld` and `ld.bfd` are hardlinked to each other.

**Comment by uberhacker** [comment]
Whatever, hardlink/straight-up copy, potatoe/potato.
_"Anyway, this should be the most fixed that it could possibly be at this point"_
You should know better than to say something like this.

**Comment by uberhacker** [comment]
@JL2210: Try it now and then install it again and see what happens.  Hopefully, you will understand what I was talking about earlier.

**Comment by JL2210** [comment]
This isn't good. You should change `if ARGV[0] == 'install'` to `def self.build` and only create the file once. Mine doesn't create a file called `ld` in the current directory, while yours still does.

**Comment by JL2210** [comment]
Hardlinks and straight up copies are not at all the same. Hardlinks have the same inode numbers as the file they link to, they save space, and they reflect the file they are linked to. Straight-up copies do not.

**Comment by uberhacker** [comment]
It's blatantly obvious you don't test.

**Comment by uberhacker** [comment]
OK, I'll give you that.  I solved the ld file residue issue.  Moving to a build function makes it harder to change the default.  Your solution requires `crew remove ld_default && crew install ld_default` whereas my solution only requires `crew install ld_default` to change the default whether the package is installed or not.

**Comment by JL2210** [comment]
Yes, I know. I'm actively working on fixing that.

**Comment by uberhacker** [comment]
No need for us to duplicate efforts.  Just provide comments to this issue.  You care too much. :)

**Comment by JL2210** [comment]
How's this?
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'file:///dev/null'
  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      $current = `find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' | cut -d/ -f2`
      if "#{$current}" == "" then
        $current = `basename $(readlink #{CREW_PREFIX}/bin/ld 2>/dev/null) 2>/dev/null`
      end
      if "#{$current}" == "" then
        $current = `cat #{CREW_PREFIX}/bin/ld 2>/dev/null | tail -1 | cut -d' ' -f1`
      end
      system "  echo \"Current default ld is #{$current}\""
    end
    puts "  Enter the new ld default:".lightblue
    puts
    puts "  b = ld.bfd".orange
    puts "  g = ld.gold".yellow
    puts "  l = ld.lld".lightgreen
    puts "  0 = Cancel".lightred
    puts

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $newdefault = "ld.bfd"
        break
      when 'g'
        $newdefault = "ld.gold"
        break
      when 'l'
        $newdefault = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "  Please select from one of the options or enter 0 to cancel.".lightred
        puts
      end
    end
    puts
    system "echo '#!/bin/bash' > ld"
    system "echo >> ld"
    system "echo '#{$newdefault} -rpath #{CREW_LIB_PREFIX} \"$@\"' >> ld"
    system "install -Dm755 ld #{CREW_PREFIX}/bin/ld"
    puts
    puts "  If you ever wish to change the default linker again, simply run `crew install ld_default`".lightblue
    puts
    abort
  end
end
```

**Comment by uberhacker** [comment]
I still want the `-L` library options because when I compile, I want it to look in `#{CREW_LIB_PREFIX}` when it's linking the shared object library files.  This also avoids the Google funky `relr.dyn` mess.

**Comment by JL2210** [comment]
Exactly. The `-rpath` option forces it to look only in `#{CREW_LIB_PREFIX}` for shared libraries and nowhere else.

**Comment by uberhacker** [comment]
But only at runtime and not compile time.

**Comment by JL2210** [comment]
All right. Agreed.

**Comment by JL2210** [comment]
```ruby
require 'package'

class Ld_default < Package
  description 'Select the default ld executable and check for libraries in #{CREW_LIB_PREFIX} first'
  homepage 'https://github.com/skycocker/chromebrew/wiki/FAQ'
  version '0.1'
  source_url 'file:///dev/null'
  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'

  depends_on 'llvm'

  def self.build
    puts
    Dir.chdir("#{CREW_PREFIX}/bin") do
      $current = `find . -inum $(ls -i ld 2>/dev/null | cut -d' ' -f1) 2>/dev/null | fgrep 'ld.' | cut -d/ -f2`
      if "#{$current}" == "" then
        $current = `basename $(readlink #{CREW_PREFIX}/bin/ld 2>/dev/null) 2>/dev/null`
      end
      if "#{$current}" == "" then
        $current = `cat #{CREW_PREFIX}/bin/ld 2>/dev/null | tail -1 | cut -d' ' -f1`
      end
      system "  echo \"Current default ld is #{$current}\""
    end
    puts "  Enter the new ld default:".lightblue
    puts
    puts "  b = ld.bfd".orange
    puts "  g = ld.gold".yellow
    puts "  l = ld.lld".lightgreen
    puts "  0 = Cancel".lightred
    puts

    while default = STDIN.gets.chomp
      case default
      when 'b'
        $newdefault = "ld.bfd"
        break
      when 'g'
        $newdefault = "ld.gold"
        break
      when 'l'
        $newdefault = "ld.lld"
        break
      when '0'
        abort
        break
      else
        puts "  Please select from one of the options or enter 0 to cancel.".lightred
        puts
      end
    end
    puts
    system "echo '#!/bin/bash' > ld"
    system "echo >> ld"
    system "echo '#{$newdefault} -rpath #{CREW_LIB_PREFIX} -L #{CREW_LIB_PREFIX} \"$@\"' >> ld"
    system "install -Dm755 ld #{CREW_PREFIX}/bin/ld"
    puts
    puts "  If you ever wish to change the default linker again, simply run `crew install ld_default`".lightblue
    puts
    abort
  end
end
```

**Comment by JL2210** [comment]
The other directories are searched anyway, so no big deal which order they come in.

**Comment by JL2210** [comment]
Since Chrome OS is so *perfect* that it doesn't have duplicate libraries in `/lib` and `/usr/lib`.

**Comment by uberhacker** [comment]
The order definitely matters because the compiler has to choose between 2 or more versions of the same file.  Trust me, this fixes the issue. :)

**Comment by JL2210** [comment]
Yeah. See the latest comment of the `.rb` file. What I was saying is that when it searches `#{CREW_LIB_PREFIX}` first, it uses any libraries it can find there instead of the system-provided ones. Then it proceeds to search `/#{ARCH_LIB}` and `/usr/#{ARCH_LIB}` because of the default search path, which it will hopefully not link against any libraries from.
TL;DR
 As long as `#{CREW_LIB_PREFIX}` is first, it's OK.

**Comment by JL2210** [comment]
@uberhacker What do you say?

**Comment by uberhacker** [comment]
I don't think the runtime option means much but it wouldn't hurt to add it.

**Comment by JL2210** [comment]
Great. So mine's OK for now?

**Comment by skiman6010** [comment]
This ready for merging now? @uberhacker

**Comment by uberhacker** [comment]
Pending your approval, sure. :)

**Comment by JL2210** [comment]
`-R` is an obsolete option. It should be `-rpath`.
